Criminal console status expansion

B-Kirill opened the pull request:
## About the PR
- Added new `Hostile` and `Eliminated` criminal console statuses

## Why / Balance
The new statuses were added for convenience:

`Hostile` status is similar to `Wanted`, but specifically marks extremely dangerous criminals. For what? It's simple. Instead of trying to remember, asking over comms, or checking a PDA to learn why someone is wanted - whether it's for minor vandalism or multiple mass-murder terrorist acts - this status lets security officers immediately recognize the threat level. At minimum, they'll be mentally prepared, denying the criminal time to escape. This is especially useful during high-pop rounds with many agents.

`Eliminated` status conveniently marks executed players or any others who shouldn't be revived. This eliminates unnecessary questions like "Is this a civilian victim or an executed criminal?" Also helpful when there are multiple corpses in the brig, wherever they came from.

## Media

Comparison:
![before](https://github.com/user-attachments/assets/255abadf-6958-4e72-b12e-905d3544553f)
![after4](https://github.com/user-attachments/assets/a5ec5c6d-54fc-41d2-a563-c2829cefe12e)


Just screenshots that everything is working as it should:
![console4](https://github.com/user-attachments/assets/3b3f0e3d-6cfb-4442-ab53-8c942a40a6b0)


## Notes
Feedback is welcome!

## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

**Changelog**
:cl: B_Kirill
add: Hostile and Eliminated criminal statuses

slarticodefast commented:
If they are twice the size of the other status icons, do they fit together with the job and health icons?
Also the mindshield icon in particular.
And afaik the art maintainers usually don't like having mixels.

B-Kirill commented:
> If they are twice the size of the other status icons, do they fit together with the job and health icons? Also the mindshield icon in particular. And afaik the art maintainers usually don't like having mixels.

It works 100% with mindshield and job icons, but it's about health.. Damn, I didn't think to check

B-Kirill commented:
Well, everything seems to be as it should be...
Although in the code I tried to make sure that their reduced rendering should not affect other icons at all.

slarticodefast commented:
That is a random, unrelated test fail. I can rerun the test for you.

B-Kirill commented:
> That is a random, unrelated test fail. I can rerun the test for you.

Oh, thanks

Pyrizhok333x2 commented:
The idea of ​​adding two new criminal statuses really sounds good.
But why can't we use the regular 8x8 icon size?
The skull image can fit there, and as for the exclamation mark, it's literally already in the criminal status, it just needs to be repainted.
Or am I missing something?

Mixelz commented:
> The idea of ​​adding two new criminal statuses really sounds good. But why can't we use the regular 8x8 icon size? The skull image can fit there, and as for the exclamation mark, it's literally already in the criminal status, it just needs to be repainted. Or am I missing something?

> If they are twice the size of the other status icons, do they fit together with the job and health icons? Also the mindshield icon in particular. And afaik the art maintainers usually don't like having mixels.

You guys are misunderstanding how the images were rendered. They were previous 8x8 images being upscaled into 24x24. 
This is why they've always looked weird or "lower quality" than the other icon images, because they actually were using less pixels and did not match the same ratio as other UI elements.
They images are now 16x16, being upscaled to 24x24, so the resulting image is much crisper and cleaner because the base image is now x2 in size.

Mixelz commented:
My only suggestion would be to maybe use a different color for the Hostile icon due to it being potentially too similar to the Wanted and thus being hard to tell at a glance.

ljm862 commented on Content.Client/StatusIcon/StatusIconOverlay.cs:
```diff
@@ -117,7 +122,17 @@ protected override void Draw(in OverlayDrawArgs args)
+                    handle.SetTransform(originalTransform);
+                }
+                else
+                {
+                    handle.DrawTexture(texture, position);
+                }
```
Here scale == 1f, do the `handle.DrawTexture` could be extracted and we can use 
`handle.DrawTexture(texture, position / scale)` regardless.

B-Kirill replied:
> Here scale == 1f, so the `handle.DrawTexture` could be extracted and we can use `handle.DrawTexture(texture, position / scale)` regardless.

I think it's pointless. If the icon has a scale flag, we will get it and run the required piece of code, otherwise we will have 1. So in any case, dividing by 1 is just an unnecessary action without profit.
(I hope I understood you correctly)

ljm862 replied:
That was my intention, yes. It's just to help minimise code duplication and improve readability, I would _hope_ the compiler would optimise away a division by 1 but we can't be sure (and I don't feel like looking at the IL code!) It's not a huge deal, I'm fine to keep it as is, just another point to be wary of :)

ljm862 commented on Content.Shared/Security/SecurityStatus.cs:
```diff
@@ -6,16 +6,20 @@
     None,
     Suspected,
     Wanted,
+    Hostile,
```
I'm always wary of adding enums in the middle of existing ones. If there's been a cast at any point from a value to an enum this will now break it. You're probably fine here, just something to be wary of!

B-Kirill replied:
> I'm always wary of adding enums in the middle of existing ones. If there's been a cast at any point from a value to an enum this will now break it. You're probably fine here, just something to be wary of!

In this case, everything is as it should be, the order in this enum determines the order in the criminal console.

slarticodefast commented:
It does seem the Urist sprite is upscaled by the same factor though. So now the icon has twice the resolution compared to that.

TeenSarlacc commented:
> My only suggestion would be to maybe use a different color for the Hostile icon due to it being potentially too similar to the Wanted and thus being hard to tell at a glance. 

animated sprite with the exclamation point flashing?

IProduceWidgets commented on Resources/Locale/en-US/criminal-records/criminal-records.ftl:
```diff
@@ -16,6 +16,8 @@ criminal-records-status-detained = Detained
 criminal-records-status-discharged = Discharged
 criminal-records-status-paroled = Paroled
+criminal-records-status-hostile = Hostile
+criminal-records-status-eliminated = Eliminated
```
```suggestion
criminal-records-status-eliminated = Unrecoverable
```
Synergizes better with the morgue.

IProduceWidgets commented on Resources/Locale/en-US/criminal-records/criminal-records.ftl:
```diff
@@ -42,6 +44,10 @@ criminal-records-console-detained = {$name} ({$job}) has been detained by {$offi
 criminal-records-console-not-parole = {$officer} cleared the parole status of {$name} ({$job}).
+criminal-records-console-hostile = {$name} ({$job}) was marked as hostile by {$officer} for: {$reason}.
+criminal-records-console-not-hostile = {$name} ({$job}) no longer marked as hostile by {$officer}.
+criminal-records-console-eliminated = {$name} ({$job}) was marked as eliminated by {$officer}.
```
```suggestion
criminal-records-console-eliminated = {$name} ({$job}) was marked as unrecoverable by {$officer}.
```

IProduceWidgets commented on Content.Shared/Security/SecurityStatus.cs:
```diff
@@ -6,16 +6,20 @@
 /// Detained - the person is detained by security
 /// Paroled - the person is on parole
 /// Discharged - the person has been released from prison
+/// Eliminated - the person has been eliminated and should not be healed
```
```suggestion
/// Unrecoverable - the person is dead and unlikely to or should not be revived.
```

B-Kirill commented:
I think it's also worth changing the Hostile icon to a slightly different color (or something else), everything else is ready in PR.

B-Kirill commented:
> I'd change either the suspicious or hostile icon for colourblind users
> 
> Don't quote me on this but there's discussion on discord whether mixels are good or bad

I think I can make a question mark for `Suspected`, that's the best I can think of. In general, I'm not that good at sprites, which explains why they look like this...

B-Kirill commented:
I'll wait until the end of the discussion.. Although something tells me that there is a 99% chance that the mixels will be rejected.

slarticodefast commented:
Yeah, I made an internal discussion thread for it and asked an art director about the mixels.
Adding the extra console status itself seems like a good addition to me though!

Mixelz commented:
> I'll wait until the end of the discussion.. Although something tells me that there is a 99% chance that the mixels will be rejected.

:(

Moryachok136 commented:
I think an icon with a picture of a target or aim would be better suited for hostile status

![image](https://github.com/user-attachments/assets/ce5966e6-382d-422c-9094-29faf5d6e5a7)

B-Kirill commented:
> I think an icon with a picture of a target or aim would be better suited for hostile status


It's not a bad idea. In this case, there is no need to change the `Suspected` icon, let the old style remain for recognition. But the scope will definitely be without a circle, even for mixels it looks terrible.

B-Kirill commented:
Updated the media and got rid of the distortion of icons in comparison

LaCumbiaDelCoronavirus commented:
Eliminated status seems a bit redundant since it frankly makes sec's job too easy, there should still be room for error with reviving people that shouldn't be revived

Hostile status is a good addition though!

B-Kirill commented:
> Eliminated status seems a bit redundant since it frankly makes sec's job too easy, there should still be room for error with reviving people that shouldn't be revived

The funny thing is that this status will only be seen by those who have a special visor or glasses. Since doctors almost never have such a thing, they may still consider some kind of corpse necessary for revival.

slarticodefast reviewed (changes requested):
Hey, I got feedback from a current art director, and from a former one.
![grafik](https://github.com/user-attachments/assets/2f2e6e6b-39f9-4c30-b3ae-913199424160)
So could you undo the texture scaling changes to the status icon code and redraw the new icons in their default 8x8 size?

B-Kirill commented:
> Hey, I got feedback from a current art director, and from a former one. ![grafik](https://private-user-images.githubusercontent.com/161409025/433395226-2f2e6e6b-39f9-4c30-b3ae-913199424160.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ2NzI4ODYsIm5iZiI6MTc0NDY3MjU4NiwicGF0aCI6Ii8xNjE0MDkwMjUvNDMzMzk1MjI2LTJmMmU2ZTZiLTM5ZjktNGMzMC1iM2FlLTkxMzE5OTQyNDE2MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxNFQyMzE2MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ZmIwY2I1Yjc1NWRjODkzOTE1NWFkMjMwOWVkNjQ0YTg1NWFkOWZmZDBiNTRkM2NjZjMzYjU4NTc3ZWVlMWI1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bmX0VssjWmCfbaivnj0BllD-c3Eb60xkLkyb6VZQfMU) So could you undo the texture scaling changes to the status icon code and redraw the new icons in their default 8x8 size?

Well, undoing the old ones will be easy, but.. I can't even theoretically imagine how I can draw a skull in 8x8 without getting a pixel mess.

slarticodefast commented:
The zombie status icon is a skull in 8x8 format as well, so maybe it makes sense to choose something more distinct from that as an icon.

B-Kirill commented:
> The zombie status icon is a skull in 8x8 format as well, so maybe it makes sense to choose something more distinct from that as an icon.

Maybe, but I still redrawn my icon and it turned out quite interesting:
![hud_eliminated](https://github.com/user-attachments/assets/a6fa903f-6718-48b4-9d49-22f34879932c)


Although I'm not sure about hostile one:
![hud_hostile](https://github.com/user-attachments/assets/29dfccb2-d25a-4cf7-9170-98ef3a470636)


I'll still try to come up with something, although I'd listen to your other ideas if you have any.

B1773rm4n commented:
**Thank you for this PR!**
The old version looks like someone just implemented the feature and said to themselves "It's a first iteration, it will do" and then it stuck forever.
Your new version is when someone just cared about to not make it look horrible. And now it's getting rejected for
"and this doesn't look great"

B-Kirill commented:
> Looks fine, though the skull icon seems to have extra pixels on top instead of a gap compared to other icons

Maybe you're right. I'll see later how it will look with the gap in the game, it was just necessary for the old icon.

B-Kirill commented:
Removed the extra pixels from the eliminated icon. I also changed the size of the icons in the PDA from 2.2 to 3.0 (they look better this way + they don't get distorted).

B1773rm4n commented:
@B-Kirill Did you revert the totally better looking icons and instead kept using the old icons? I was really looking forward to your change. But I see you removed the old screenshot. Can you please keep it somewhere in this thread so other people will have context?

B-Kirill commented:
> @B-Kirill Did you revert the totally better looking icons and instead kept using the old icons? I was really looking forward to your change. But I see you removed the old screenshot.

Yes, it's sad, but I can't argue with two art directors at once (even if one is a former one), otherwise this PR simply won't be accepted.

> Can you please keep it somewhere in this thread so other people will have context?

Okay, I can upload it in the comments.

B-Kirill commented:
### Comparison with rejected icon variations

![image](https://github.com/user-attachments/assets/5b9766a3-ae29-4fd8-9eaf-9ae88a7d37f4)

![image](https://github.com/user-attachments/assets/63d79b8c-9617-4d3c-af9c-26c2594a8c80)

slarticodefast commented on Resources/Textures/Interface/Misc/security_icons.rsi/meta.json:
```diff
@@ -1,7 +1,7 @@
   "version": 1,
   "license": "CC-BY-SA-3.0",
-  "copyright": "Taken from https://github.com/tgstation/tgstation/blob/7f654b6e8e59021607a9e888dfeb79920401c372/icons/mob/huds/hud.dmi",
+  "copyright": "Original sprites from https://github.com/tgstation/tgstation/blob/7f654b6e8e59021607a9e888dfeb79920401c372/icons/mob/huds/hud.dmi. Extended with new status sprites by B_Kirill",
```
```suggestion
  "copyright": "Original sprites from https://github.com/tgstation/tgstation/blob/7f654b6e8e59021607a9e888dfeb79920401c372/icons/mob/huds/hud.dmi. hud_hostile and hud_eliminated by B_Kirill",
```

slarticodefast commented on Content.Client/CartridgeLoader/Cartridges/WantedListUiFragment.cs:
```diff
@@ -201,7 +201,7 @@ private void GenerateItem(ListData data, ListContainerButton button)
         var rect = new TextureRect()
         {
-            TextureScale = new(2.2f),
+            TextureScale = new(3.0f),
```
I checked both versions in-game and the original value looks better to me since the icon does not go up to the border of the bounding box. Also the change is kinda unrelated to this PR, so I'll change it back for now.
